lib/pavilion/builder.py

```diff
@@ -567,8 +575,10 @@ def _build(self, build_dir, cancel_event, lock: lockfile.LockFile = None):
                             return False
 
         except subprocess.CalledProcessError as err:
-            if cancel_event is not None:
+            if cancel_event is not None and hard_fail:
                 cancel_event.set()
```

Comment:
Only the master thread needs to know about hard fail and set the cancel event. It should be watching for the the fail event, and then use that to decide whether to cancel.